pytest_invenio/fixtures.py

```diff
@@ -633,3 +634,77 @@ def create_bucket_from_dir(source_dir, location_obj=None):
         db.session.commit()
         return bucket_obj
     return create_bucket_from_dir
+
+
+def _noop(*args, **kwargs):
+    pass
+
+
+def _iter_entry_points(real_iter, extras, group, name=None):
+    """Helper to add in extra entry points."""
+    # First iter through existing entry points
+    for i in real_iter(group, name=name):
+        yield i
+    # Then add extras if defined.
+    if group in extras:
+        for ep_str in extras[group]:
+            ep = pkg_resources.EntryPoint.parse(ep_str)
+            # Mock required, because we don't have distribution.
+            ep.require = _noop
+            yield ep
+
+
+@pytest.fixture(scope="module")
+def entry_points(extra_entry_points):
+    """Entry points fixture.
+
+    Scope: module
+
+    Invenio relies heavily on Python entry points for constructing an
+    application and it can be rather cumbersome to try to register database
+    models, search mappings etc yourself afterwards.
+
+    This fixture allows you to inject extra entry points into the application
+    loading, so that you can load e.g. a testing module or test mapping.
+
+    To use the fixture simply define the ``extra_entry_points()`` fixture,
+    and then depend on the ``entry_points()`` fixture in your ``create_app``
+    fixture:
+
+    .. code-block:: python
+
+        @pytest.fixture(scope="module")
+        def extra_entry_points():
+            return {
+                'invenio_db.models': [
+                    'mock_module = mock_module.models',
+                ]
+            }
+
+        @pytest.fixture(scope="module")
+        def create_app(instance_path, entry_points):
+            return _create_api
+    """
+    # Save reference to existing function
+    real_iter = pkg_resources.iter_entry_points
+
+    # Mock existing function
+    pkg_resources.iter_entry_points = partial(
```

Comment:
This only works when used like this:

```
import pkg_resources
pkg_resources.iter_entry_points(..)
```

Doesn't work when a module holds a reference first:

```
from pkg_resources import iter_entry_points
iter_entry_points(..)
```